provider/openstack/errors.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package openstack
+
+import (
+	gooseerrors "gopkg.in/goose.v2/errors"
+
+	"github.com/juju/juju/environs/context"
+)
+
+func MaybeHandleCredentialError(err error, ctx context.ProviderCallContext) (error, bool) {
+	isUnauthorized := gooseerrors.IsUnauthorised(err)
+	//fmt.Printf("XXXXX... IsUnauthorized : %v\n", IsUnauthorized)
+	if ctx != nil && isUnauthorized {
```

Comment:
If unauthorised, also need to wrap err with common.CredentialNotValid
See how the ec2 provider does it